tests/Mono.Android-Tests/Xamarin.Android.Net/AndroidMessageHandlerTests.cs

```diff
@@ -111,6 +111,25 @@ public async Task ServerCertificateCustomValidationCallback_IgnoresCertificateHo
 			Assert.AreEqual (SslPolicyErrors.RemoteCertificateNameMismatch, reportedErrors & SslPolicyErrors.RemoteCertificateNameMismatch);
 		}
 
+		[Test]
+		public async Task ServerCertificateCustomValidationCallback_Redirects ()
+		{
+			int callbackCounter = 0;
+
+			var handler = new AndroidMessageHandler {
+				ServerCertificateCustomValidationCallback = (request, cert, chain, errors) => {
+					callbackCounter++;
+					return errors == SslPolicyErrors.None;
+				}
+			};
+
+			var client = new HttpClient (handler);
+			var result = await client.GetAsync ("https://httpbin.org/redirect-to?url=https://www.microsoft.com/");
+
+			Assert.AreEqual (2, callbackCounter);
```

Comment:
@simonrozsival: why is this `2`?  Are there any docs around how many times `ServerCertificateCustomValidationCallback` will be invoked, other than "at least once"?  Why is the callback being called twice?  Is that a bug?  Is that likely to change at any point in the future?

Or *should* it be 2: once for `httpbin.org/redirect-to`, and once again for `www.microsoft.com`, and it's called "once per redirect", i.e. if you "nested" redirects with `https://httpbin.org/redirect-to?url=https://httpbin.org/redirect-to?url=…`, `callbackCounter` should be the number of times we hit `httpbin.org/redirect-to` + 1?

Reply:
The callback is called once for every sub-request that's made (the first request + all redirects). In this case, it's called once for `httpbin.org/...` and once for `www.microsoft.com`. So `callbackCounter` should equal to 1 + number of redirects. The `SocketsHttpHandler` behaves the same way.